### ISSUE TITLE
Hazelcast not handling XML config attribute for DurableExecutorService.

### ISSUE THREAD

**Issue by rscnt** [hazelcast/hazelcast#9078]
Hazelcast version: 3.7.2
Cluster size: 1

Java

```
java version "1.8.0_91" 
Java(TM) SE Runtime Environment (build 1.8.0_91-b14) 
Java HotSpot(TM) 64-Bit Server VM (build 25.91-b14, mixed mode) 
```

OS: Linux 4.2.3-300.fc23.x86_64 

hazelcast.xml:  [hazelcast.xml](https://github.com/hazelcast/hazelcast/files/519916/hazelcast.xml.txt)

The configuration is written [as it's indicated in the documentation](http://docs.hazelcast.org/docs/3.7/manual/html-single/index.html#configuring-durable-executor-service), however it seems that hazelcast is ignoring/not handling, the `<durable-executor-service />` nodes.

These are the only things I noticed peeking a little in the code that's on master:

The hazelcast-full-example.xml is using `<executor-service />` for the `<durable-executor-service />` example.

https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/resources/hazelcast-full-example.xml#L429

And the ConfigXMLGenerator includes a `<durable-executor-service />` node:

https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/config/ConfigXmlGenerator.java#L123

But if I'm understand correctly, since the `<durable-executor-service />` node is not present on the examples the test:

https://github.com/hazelcast/hazelcast/blob/05f3a177d748740fa2d2f68544d5ee5408b46836/hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java#L290

Does not check if it's "valid" or not.

XMLConfigBuilder#handleXMLNode is not "handling"  the `<durable-executor-service / >`:

https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/config/XmlConfigBuilder.java#L275

And the node is not declared in XMLElements enum:

https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/config/XmlElements.java#L19

I can't write an unit test for it right now, also it's probably that you are all aware of this issue, but I couldn't find a reference for it anywhere.

**Comment by mmedenjak** [comment]
Hi @rscnt , great analysis, you are right about everything. All of this has been fixed, including the spring configuration and will be included in the next release.
Thanks for the effort!

**Comment by jerrinot** [comment]
closing it's already fixed in 3.8
@rscnt: once more thanks for this great report!
